### PR TITLE
Add/Update Donation goal meta on donation form update or when donating #2250

### LINKED ISSUE #2250: feat(form): allow forms to be queried based on close status
## Issue Overview
Occasionally, users want to have an archive of Give forms, but exclude those that are closed because they achieved their goal. We currently have `post_meta` called `_give_close_form_when_goal_achieved` which returns just `enabled|disabled`, but when the goal is actually achieved and the form is closed, there's no way to query and exclude those forms. Adding `_give_form_is_closed` with `true|false` would benefit these users greatly.

### PR DESCRIPTION
**Pull request by raftaar1191**
## Description
PR to fix #2250 

## How Has This Been Tested?
Manual tested
Please see the video to know how this is being tested
Video Link: https://screencast-o-matic.com/watch/cFV303o3Lh

## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: includes/payments/actions.php [hunk 1]
```diff
@@ -449,4 +449,44 @@ function give_payment_save_page_data( $payment_id ) {
 }
 
 // Fire when payment is save.
-add_action( 'give_insert_payment', 'give_payment_save_page_data' );
\ No newline at end of file
+add_action( 'give_insert_payment', 'give_payment_save_page_data' );
+
+/**
+ * Add/Update goal closed meta when donation form is update.
```
Comment (DevinWalker) [inline includes/payments/actions.php]: Typo: `is updated.`

Comment (raftaar1191) [reply includes/payments/actions.php]: Fixed

**Comment by raftaar1191** [comment]
Thanks, @ravinderk for reviewing the PR. I am working on **Donation Status Change** Case as well

**Comment by raftaar1191** [comment]
@ravinderk Fix when Admin update the **Donation Status** from bulk action and from the Donation Single Page

**Comment by raftaar1191** [comment]
@ravinderk 

Add the DB upgrade please check and let me know if this is fine or not

**Comment by ravinderk** [comment]
@raftaar1191 Create this pr again.

**Pull request merged**